pkg/project/project.go

```diff
@@ -278,6 +278,9 @@ func (p *Project) Environments() string { return p.projectfile.Environments }
 
 // New creates a new Project struct
 func New(p *projectfile.Project, out output.Outputer) (*Project, error) {
+	if p == nil {
+		return nil, locale.NewError("err_project_projectfile", "Cannot create new project without a projectfile")
+	}
```

Comment:
I'm not certain that we should be policing the input of this function but this change will ensure that we do not have projects without projectfiles